miden/tests/integration/stdlib/collections/smt.rs

```diff
@@ -0,0 +1,191 @@
+
+    // compute pseudo-random key/value pair
+    let key = <[u8; 8]>::try_from(&seed.as_bytes()[..8]).unwrap();
+    let key = u64::from_le_bytes(key);
+    let key = [Felt::new(key); WORD_SIZE];
+    let value = <[u8; 8]>::try_from(&seed.as_bytes()[8..16]).unwrap();
+    let value = u64::from_le_bytes(value);
+    let value = [Felt::new(value); WORD_SIZE];
+
+    fn assert_case(
+        depth: u64,
+        index: u64,
+        key: Word,
+        value: Word,
+        node: Word,
+        root: Word,
+        store: MerkleStore,
+    ) {
+        let source = r#"
+            begin
+                adv.smtget
+                dropw
+                dropw
+                adv_push.10
+            end
+        "#;
+        let initial_stack = [
+            root[0].as_int(),
+            root[1].as_int(),
+            root[2].as_int(),
+            root[3].as_int(),
+            key[0].as_int(),
+            key[1].as_int(),
+            key[2].as_int(),
+            key[3].as_int(),
+        ];
+        let expected_output = [
+            node[3].as_int(),
+            node[2].as_int(),
+            node[1].as_int(),
+            node[0].as_int(),
+            value[3].as_int(),
+            value[2].as_int(),
+            value[1].as_int(),
+            value[0].as_int(),
+            depth,
+            index,
+        ];
+        let advice_stack = [];
+        let advice_map = [(key.to_owned().into_bytes(), value.to_vec())];
+        build_test!(source, &initial_stack, &advice_stack, store.clone(), advice_map)
+            .expect_stack(&expected_output);
+
+        let source = r#"
+            use.std::collections::smt
+
+            begin
+                exec.smt::get
+            end
+        "#;
+        let initial_stack = [
+            root[0].as_int(),
+            root[1].as_int(),
+            root[2].as_int(),
+            root[3].as_int(),
+            key[0].as_int(),
+            key[1].as_int(),
+            key[2].as_int(),
+            key[3].as_int(),
+        ];
+        let expected_output = [
+            depth,
+            index,
+            value[3].as_int(),
+            value[2].as_int(),
+            value[1].as_int(),
+            value[0].as_int(),
+            key[3].as_int(),
+            key[2].as_int(),
+            key[1].as_int(),
+            key[0].as_int(),
+            root[3].as_int(),
+            root[2].as_int(),
+            root[1].as_int(),
+            root[0].as_int(),
+        ];
+        let advice_stack = [];
+        let advice_map = [(key.to_owned().into_bytes(), value.to_vec())];
+        build_test!(source, &initial_stack, &advice_stack, store, advice_map)
+            .expect_stack(&expected_output);
+    }
+
+    // insert the leaf on the first tier
+    let depth = 16;
+    let index = key[3].as_int() >> 48;
+    let mut store = MerkleStore::new();
+    let node = Rpo256::merge_in_domain(&[key.into(), value.into()], Felt::new(depth)).into();
```

Comment:
This is not quite correct (or at least not what we've been discussing). The node value should be computed as `hash(remaining_key, value)` and not as `hash(key, value)` right now.

Reply:
True, will fix it here
https://github.com/0xPolygonMiden/miden-vm/pull/808#discussion_r1154013803